### PR TITLE
Subsscription module for OVS

### PR DESCRIPTION
**Pull request by douradofrazer**
Resolves:
https://dcsa.atlassian.net/browse/DDT-113

### CODE REVIEW

File: src/main/java/org/dcsa/ovs/model/transferobjects/OVSEventSubscriptionTO.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  private String transportCallID;
+
+  private String vesselIMONumber;
+
+  private String carrierVoyageNumber;
+
+  private String carrierServiceCode;
+
+  private String operationsEventTypeCode;
```
Comment (nt-as) [inline src/main/java/org/dcsa/ovs/model/transferobjects/OVSEventSubscriptionTO.java:25]: Should this be a `List<String>` (to match the rewrite we are doing for other event type code).

File: src/main/java/org/dcsa/ovs/service/impl/OVSEventSubscriptionTOServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+import org.dcsa.core.events.model.base.AbstractEventSubscription;
+import org.dcsa.core.events.model.enums.*;
+import org.dcsa.core.events.repository.EventSubscriptionRepository;
+import org.dcsa.core.events.service.EventSubscriptionService;
+import org.dcsa.core.events.service.impl.EventSubscriptionTOServiceImpl;
+import org.dcsa.core.exception.UpdateException;
+import org.dcsa.core.util.MappingUtils;
+import org.dcsa.ovs.model.transferobjects.OVSEventSubscriptionTO;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
```
Comment (nt-as) [inline src/main/java/org/dcsa/ovs/service/impl/OVSEventSubscriptionTOServiceImpl.java:14]: The class says `Mainly for internal use within the framework; consider Apache's Common Lang for ...`

**Review by nt-as — COMMENTED** [review]
(Quick drive by review)

File: src/main/java/org/dcsa/ovs/service/impl/OVSEventSubscriptionTOServiceImpl.java (hunk 2 above)
Comment (douradofrazer) [reply src/main/java/org/dcsa/ovs/service/impl/OVSEventSubscriptionTOServiceImpl.java]: I looked into , you're right. Though the implementation is a simple one , its better to use common lang. However not required anymore , so removed it. Will keep in mind for when I need to use it next time.

File: src/main/java/org/dcsa/ovs/service/impl/OVSEventSubscriptionTOServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+  public Mono<OVSEventSubscriptionTO> update(OVSEventSubscriptionTO eventSubscriptionTO) {
+    if (eventSubscriptionTO.getSecret() != null) {
+      return Mono.error(
+          new UpdateException(
+              "Please omit the \"secret\" attribute.  If you want to change the"
+                  + " secret, please use the dedicated secret endpoint"
+                  + " (\"PUT .../event-subscriptions/"
+                  + eventSubscriptionTO.getSubscriptionID()
+                  + "/secret\")."));
+    }
```
Comment (nt-as) [inline src/main/java/org/dcsa/ovs/service/impl/OVSEventSubscriptionTOServiceImpl.java:124]: Same remark as for TNT (this should be handled in the subclass somehow)

Comment (douradofrazer) [reply src/main/java/org/dcsa/ovs/service/impl/OVSEventSubscriptionTOServiceImpl.java]: Added hook.

**Review by HenrikHL — APPROVED** [review]

**Pull request merged**